src/MinimalDelegation.sol

```diff
@@ -67,4 +80,19 @@ contract MinimalDelegation is IERC7821 {
             if (!success) revert IERC7821.CallFailed();
         }
     }
+
+    function _authorize(Key memory key) private returns (bytes32 keyHash) {
+        keyHash = key.hash();
+        MinimalDelegationStorage storage minimalDelegationStorage = MinimalDelegationStorageLib.get();
+        minimalDelegationStorage.keyStorage[keyHash] = abi.encode(key);
+        minimalDelegationStorage.keyHashes.add(keyHash);
```

Comment:
check the result from .add()? I think it returns a bool

Reply:
this would return false though if its the key already exists with a different expiry

Reply:
i think the key hash is only over the keyType + public key, maybe we should change that

Reply:
would it be weird to have 2 keys authorized with only different expiries? unless they revoke the old one first

Reply:
i think the current behavior is correct but feel free to add a comment maybe?